backend/src/data/repositories/chat-message/chat-message.repository.ts

```diff
@@ -0,0 +1,72 @@
+import { SortOrder } from '~/common/enums/enums';
+import {
+  ChatGetAllMessagesRequestDto,
+  ChatGetLastMessagesRequestDto,
+  ChatMessageCreateRequestDto,
+  ChatMessageGetAllItemResponseDto,
+} from '~/common/types/types';
+import { ChatMessage as ChatMessageM } from '~/data/models/models';
+
+type Constructor = {
+  ChatMessageModel: typeof ChatMessageM;
+};
+
+class ChatMessage {
+  #ChatMessageModel: typeof ChatMessageM;
+
+  public constructor({ ChatMessageModel }: Constructor) {
+    this.#ChatMessageModel = ChatMessageModel;
+  }
+
+  public getAll({
+    chatId,
+  }: ChatGetAllMessagesRequestDto): Promise<
+    ChatMessageGetAllItemResponseDto[]
+  > {
+    return this.#ChatMessageModel
+      .query()
+      .select()
+      .where({ chatId })
+      .withGraphJoined(
+        '[sender(withoutPassword).[userDetails], receiver(withoutPassword).[userDetails]]',
+      )
+      .castTo<ChatMessageGetAllItemResponseDto[]>()
+      .execute();
+  }
+
+  public create({
+    senderId,
+    receiverId,
+    message,
+    chatId,
+  }: ChatMessageCreateRequestDto): Promise<ChatMessageGetAllItemResponseDto> {
```

Comment:
```suggestion
  }: DeepNonNullable<ChatMessageCreateRequestDto>): Promise<ChatMessageGetAllItemResponseDto> {
```
wanna try? 😃 
[playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAIhFgHIHsB2iCuAbLBDARlhADwAqAfFALxQDeAUAJADaA0lAJapQDWEIyAGZRSAXQBcseEjSYcBIsRTpseQiVJtR5cvQC+9eqEhQAwgAtcwALIQAzndwBzCKYBOEKxABKEAI4Y9sAwwMjUdPRQUB4AxhAcAG4QbgCSACaSqBgAtvjJANyRUHYQqGnJ6Zk5eW6FUdn2ji6SdsBuXE51UDGWwJXFbR1QAD5QWThd+LgAXpIWVrYOzq4eXr4BQSFho+NY+oX0MWitUFNuknAIynJqivM2jcvunsA+-oGtW5Q0DIyx8UlUhkAIwAGiYJTKFQyUDBTAaS2aUAA5IJkGEzsjwYwelZ+sjUMhgGNVFimFNZhFGH8IHFEtDJAAmbGMSHlIFMlkIpoQSTIinFMC4bJk6m4vow3ZMAx6IA)

Reply:
But that chatId can be null, if it is a first chat message between users

Reply:
It can null in repository? 
in [migration](https://github.com/BinaryStudioAcademy/bsa-2022-guruhub/blob/dd298348d179ec37170279cc358a2ed29ea24913/backend/src/data/migrations/20220830154032_add_chat_messages_table.ts#L44) it's nonNullable

Reply:
Oh, okay, agree 😄